backtracking/sudoku_solve.cpp

```diff
@@ -1,91 +1,144 @@
+/**
+ * @file
+ * @brief [Sudoku Solver](https://en.wikipedia.org/wiki/Sudoku) algorithm.
+ *
+ * @details
+ * Sudoku (数独, sūdoku, digit-single) (/suːˈdoʊkuː/, /-ˈdɒk-/, /sə-/, originally called
+ * Number Place) is a logic-based, combinatorial number-placement puzzle. 
+ * In classic sudoku, the objective is to fill a 9×9 grid with digits so that each column,
+ * each row, and each of the nine 3×3 subgrids that compose the grid (also called "boxes", "blocks", or "regions")
+ * contain all of the digits from 1 to 9. The puzzle setter provides a
+ * partially completed grid, which for a well-posed puzzle has a single solution.
+ *
+ * @author [DarthCoder3200](https://github.com/DarthCoder3200)
+ * @author [David Leal](https://github.com/Panquesito7)
+ */
 #include <iostream>
-using namespace std;
-/// N=9;
-int n = 9;
+#include <array>
 
-bool isPossible(int mat[][9], int i, int j, int no) {
-    /// Row or col nahin hona chahiye
-    for (int x = 0; x < n; x++) {
-        if (mat[x][j] == no || mat[i][x] == no) {
-            return false;
+/**
+ * @namespace backtracking
+ * @brief Backtracking algorithms
+ */
+namespace backtracking {
+    /**
+     * Checks if it's possible to place a 'no'
+     * @tparam V number of vertices in the array
+     * @param mat matrix where numbers are saved
+     * @param i current index in rows
+     * @param j current index in columns
+     * @param no number to be added in matrix

```

Comment:
I think this is the only thing that is missing good documentation.
Is there anything else that needs to be changed/fixed?